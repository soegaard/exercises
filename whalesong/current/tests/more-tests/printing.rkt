#lang planet dyoo/whalesong

(current-print-mode "constructor")


'(1 2 3)
(list "hello" "world")
(list 'hello 'world)
(cons 1 2)
(cons 1 (cons 2 (cons 3 empty)))
(cons 1 (cons 2 (cons 3 4)))
'()

'hello
(box 'hello)
(vector 'hello 'world)

(define-struct person (name age))
(person 'danny 32)
(person "jerry" 32)


;; This is slightly broken: we should follow DrRacket shared printing
;; notation.
(shared ([a (cons 1 a)])
  a)