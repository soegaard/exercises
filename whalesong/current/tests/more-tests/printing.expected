(list 1 2 3)
(list "hello" "world")
(list 'hello 'world)
(cons 1 2)
(list 1 2 3)
(cons 1 (cons 2 (cons 3 4)))
(list)
'hello
(box 'hello)
(vector 'hello 'world)
(person 'danny 32)
(person "jerry" 32)
#0=(cons 1 #0#)
